### PR TITLE
Disallow adding fields to annotation declarations

### LINKED ISSUE #676: It should not be possible to add a Field to an AnnotationDeclaration
The methods `addField` should be overridden in `AnnotationDeclaration` to return `UnsupportedOperationException`

### PR DESCRIPTION
**Pull request by matozoid**
Fixes #676.

### CODE REVIEW

File: javaparser-core-testing/src/test/java/com/github/javaparser/ast/body/AnnotationDeclarationTest.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class AnnotationDeclarationTest {
+    @Test
+    void cantAddField() {
+        assertThrows(IllegalStateException.class, () -> new AnnotationDeclaration().addField(Object.class, "oo"));
+    }
+}
```
Comment (ftomassetti) [inline javaparser-core-testing/src/test/java/com/github/javaparser/ast/body/AnnotationDeclarationTest.java]: nitpick: new line at the end of the file
```suggestion
}

```

File: javaparser-core/src/main/java/com/github/javaparser/ast/body/AnnotationDeclaration.java [hunk 2]
```diff
@@ -135,4 +136,9 @@ public ResolvedAnnotationDeclaration resolve() {
     public Optional<AnnotationDeclaration> toAnnotationDeclaration() {
         return Optional.of(this);
     }
+
+    @Override
+    public FieldDeclaration addField(Type type, String name, Modifier.Keyword... modifiers) {
+        throw new IllegalStateException("Cannot add a field to an annotation declaration.");
```
Comment (ftomassetti) [inline javaparser-core/src/main/java/com/github/javaparser/ast/body/AnnotationDeclaration.java:142]: This seems a good idea

**Review by ftomassetti — APPROVED** [review]
+1

File: javaparser-core/src/main/java/com/github/javaparser/ast/body/AnnotationDeclaration.java (hunk 2 above)
Comment (matozoid) [reply javaparser-core/src/main/java/com/github/javaparser/ast/body/AnnotationDeclaration.java:142]: @ftomassetti you suggested it years ago, see issue :-)

Comment (ftomassetti) [reply javaparser-core/src/main/java/com/github/javaparser/ast/body/AnnotationDeclaration.java:142]: I was so wise when I was younger :)

**Pull request merged**